finalitygadget/finalitygadget.go

```diff
@@ -384,8 +385,14 @@ func (fg *FinalityGadget) QueryLatestFinalizedBlock() (*types.Block, error) {
 	return fg.db.QueryLatestFinalizedBlock()
 }
 
-// This function process blocks indefinitely, starting from the last finalized block.
-func (fg *FinalityGadget) ProcessBlocks(ctx context.Context) error {
+// This function is run once at startup and starts the FG from the last finalized block.
```

Comment:
if this is run once, do we still need the logic "Start polling for new blocks at set interval" below?

Reply:
yes both loops are necessary:
- startup loop runs until btc staking is activated
- block processing loop runs to fetch new blocks

Reply:
got it. so the `ProcessBlocks` go routine won't start until btc staking is activated?